From 0eae85556558b410635ad03ed5eccb9648e11fce Mon Sep 17 00:00:00 2001
From: Hans de Goede <hdegoede@redhat.com>
Date: Wed, 29 Jan 2025 19:26:16 +0100
Subject: [PATCH 12/12] usbio: Use MAX_PAYLOAD_BSIZE in usbio_bulk_write()

usbio_bulk_write() uses a struct usbio_bmsg header rather than
a struct usbio_msg header. So its max payload length check must use
MAX_PAYLOAD_BSIZE instead of MAX_PAYLOAD_SIZE.

Signed-off-by: Hans de Goede <hdegoede@redhat.com>
---
 drivers/mfd/usbio.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mfd/usbio.c b/drivers/mfd/usbio.c
index 51757df..ab5dae9 100644
--- a/drivers/mfd/usbio.c
+++ b/drivers/mfd/usbio.c
@@ -303,7 +303,7 @@ static int usbio_bulk_write(struct usbio_stub *stub, u8 cmd, const void *obuf,
 	if (bridge->state == BRIDGE_STOPPED)
 		return -ENODEV;
 
-	if (obuf_len > MAX_PAYLOAD_SIZE)
+	if (obuf_len > MAX_PAYLOAD_BSIZE)
 		return -EINVAL;
 
 	if (last_pkt)
-- 
2.47.1

